ament_tools/build_types/bazel.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+
+from ament_tools.build_type import BuildAction
+from ament_tools.build_type import BuildType
+
+from ament_tools.context import ContextExtender
+
+from ament_tools.helper import extract_argument_group
+
+
+def _has_target(path, target):
+    # To query bazel for a list of all labels, we use:
+    # bazel query '//...' --output=label
+    output = subprocess.check_output(['bazel', 'query', '//...', '--output=label'], cwd=path)
```

Comment:
I would suggest to define the `bazel` executable once as this package does for e.g. CMake and use the global variable everywhere: https://github.com/ament/ament_tools/blob/63ec41d7ff942295e52235bae07a91a13a439420/ament_tools/build_types/cmake_common.py#L21

Reply:
Should be fixed in d4ca246